### DISCUSSION TITLE
how to properly filter ec2 instance import using filter by instance name ?  and by id?

### DISCUSSION THREAD

**Discussion by AlexandreHGuerra** [GoogleCloudPlatform/terraformer#1110] [General]
tried many times with filter definitions with no avail...  all many imports got all instances everytime

**Comment by sunil29feb** [comment]
Hi @alexhguerra  ,

Please try like this -

1. For instance name -
terraformer import aws --resources=ec2_instance --filter="Name=tags.Name;Value=Terraformer-xyz" --regions=us-east-1

2. For instance id 
terraformer import aws --resources=ec2_instance --filter="Name=id;Value=i-0xxxxxxxxx" --regions=us-east-1

Regards,
Sunil

**Reply by adrianomatildes** [reply]
Hello, @sunil29feb 
doesn't recognize a --resources flag

**Reply by Pranali-Mande** [reply]
Hi @sunil29feb ,
Can you please let me know how can we pass different filter params for different resources. Like in below command I am able to filter by passing tags and id for both ec2_instance and vpc

./terraformer import aws --resources=ec2_instance,vpc --filter="Name=tags.Name;Value=web-server:production" --filter="Name=id;Value=i-xxxxxx:vpc-0aaxxxxxch3f805" —connect=true --regions=us-east-1 --profile=""

But I want to only filter ec2_instance by tag while the vpc tag can be anything and vpc should be filtered by the id. I tried the following way, but it returned 0 resources, though there were resources satisfying below condition

./terraformer import aws --resources=ec2_instance --filter="Name=tags.Name;Value=web-server" --resources=vpc --filter="Name=id;Value=vpc-0aaXXXXXX5" connect=true --regions=us-east-1 --profile=""

Can you please let me know how should the command be created?

**Comment by sunil29feb** [comment]
@adrianomatildes - Which version of terraformer you are using?

**Reply by adrianomatildes** [reply]
The version is this: Terraform v1.0.11

**Reply by adrianomatildes** [reply]
@sunil29feb 
Sorry, follow the correct information
Terraformer v0.8.18

**Reply by sunil29feb** [reply]
@adrianomatildes - Please share the error message and a screenshot of the terminal.
